clients/ui/bff/internal/mocks/k8s_mock.go

```diff
@@ -2,45 +2,210 @@ package mocks
 
 import (
 	"context"
+	"fmt"
 	k8s "github.com/kubeflow/model-registry/ui/bff/internal/integrations"
-	"github.com/stretchr/testify/mock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
 	"log/slog"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
 type KubernetesClientMock struct {
-	mock.Mock
+	*k8s.KubernetesClient
+	testEnv *envtest.Environment
 }
 
 func (m *KubernetesClientMock) Shutdown(ctx context.Context, logger *slog.Logger) error {
-	logger.Error("Shutdown was called in mock")
+	logger.Info("Shutdown was called in mock")
+	m.StopFn()
+	err := m.testEnv.Stop()
+	if err != nil {
+		logger.Error("timeout while waiting for Kubernetes manager to stop")
+		return fmt.Errorf("timeout while waiting for Kubernetes manager to stop")
+	}
+	logger.Info("Shutdown ended successfully")
 	return nil
 }
 
-func NewKubernetesClient(_ *slog.Logger) (k8s.KubernetesClientInterface, error) {
-	return &KubernetesClientMock{}, nil
+func NewKubernetesClient(logger *slog.Logger, ctx context.Context, cancel context.CancelFunc) (k8s.KubernetesClientInterface, error) {
+
+	projectRoot, err := getProjectRoot()
+	if err != nil {
+		logger.Error("failed to find project root to locate binaries", err)
```

Comment:
go vet was failing on this line and the other `logger.Error` lines due to:
```
slog.Logger.Error arg "err" should be a string or a slog.Attr (possible missing key or value)
```

Reply:
Fixed! But I don't know why I didn't get this error locally! :P